### ISSUE TITLE
Caching icons

### ISSUE THREAD

**Issue by cheater** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#35]
Icon: allow caching external URL in data: scheme to cache it; add option to always fall back to the cache when URL is not available; add option to asynchronously update the icon once every n seconds (set default to 7 days) - this update should happen in the background, so that you don't have to wait for the search engine icons to show (this is an issue that sometimes happens in SSS)

Also please add an option to convert a http/https URL to a data: URL. This means I paste a URL to an icon in the icon box (for example `https://www.google.com/favicon.ico`, and then I press a button, and the URL gets replaced with the `data: ...` representation.

**Comment by YoshifumiFuyuno** [comment]
I don't understand what you are saying.
The following is the default behavior.

If you add a search engine from the context menu or options page, cache the favicon as data: scheme.

**Comment by cheater** [comment]
I've noticed with SSS that sometimes when I open the search menu for the first time after some time, the icons will take some time to load. The menu is displayed without icons, and then they load in. I assumed it was the same with SPC, it being a closely related fork. So I reported the issue here. Can it be that the cache can become invalid (time out), and when you open the menu, the icons will be loaded from the web before they are displayed?
